task6__israel_jacob/src/Cities.cpp

```diff
@@ -0,0 +1,161 @@
+#pragma once
+#include "Cities.h"
+#include <iostream>
+#include <fstream>
+
+#include <cstdlib>
+
+// Constructor
+Cities::Cities() {
+	cities = std::map<std::string, Location>();
+}
+
+// Helper method to check file validity and return the name of the first city
+std::string Cities::checkFile(std::ifstream& inputFile) {
+	if (!inputFile.is_open()) {
+		throw std::exception("Unable to open file.");
+	}
+
+	std::string name;
+
+	if (!std::getline(inputFile, name)) {
+		inputFile.close();
+		throw std::exception("The file is empty");
+	}
+	return name;
+}
+
+// Method to add cities from a file
+void Cities::addFromFile(std::string path) {
+	std::ifstream inputFile(path);
+
+	std::string name;
+	std::string x, y;
+
+	try {
+		name = checkFile(inputFile);
+	}
+	catch (std::exception& e) {
+		throw e;
+	}
+
+	do {
+		std::getline(inputFile, x, '-');
+		std::getline(inputFile, y);
+		try {
+			Location loc = Location(stold(x), stold(y));
+			cities.emplace(name, loc);
+		}
+		catch (std::exception& e) {
+			inputFile.close();
+			throw std::exception("The information is invalid");
+		}
+
+	} while (std::getline(inputFile, name));
+	inputFile.close();
+}
+
+// Method to check if a city exists
+bool Cities::exists(std::string name) {
+	auto i = cities.find(name);
+	return cities.find(name) != cities.end();
+}
+
+// Method to find cities near a specified city within a given radius using a specified distance norm
+std::vector<std::pair<std::string, Location>> Cities::nearCities(std::string name, double radius, int norm) {
+	auto cityIterator = cities.find(name);
+
+	double x = cityIterator->second.getX();
+	double y = cityIterator->second.getY();
+	std::vector<std::pair<std::string, Location>> nearCities = distanceBySquare(radius, x, y);
+
+	if (norm == 0)
```

Comment:
you shouldn't use if else, you should use lambda expression and save the functions in a map

Reply:
why?